### PR TITLE
operator [N] [CI] self-node-remediation (0.4.0)

### PR DESCRIPTION
**Pull request by mshitrit**
This is the first version of the self node remediation operator which is replacing the [poison pill ](https://github.com/redhat-openshift-ecosystem/community-operators-prod/tree/main/operators/poison-pill-operator) operator.

### CODE REVIEW

**Comment by J0zi** [comment]
@mshitrit it is failing when package name 'self-node-remediation' is not same as operator dir name 'self-node-remediation-operator.

File: operators/self-node-remediation/0.4.0/metadata/annotations.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  operators.operatorframework.io.metrics.builder: operator-sdk-v1.18.0+git
+  operators.operatorframework.io.metrics.mediatype.v1: metrics+v1
+  operators.operatorframework.io.metrics.project_layout: go.kubebuilder.io/v3
+
+  # Annotations for OCP
+  com.redhat.openshift.versions: "v4.11"
+
+  # Annotations for testing.
+  operators.operatorframework.io.test.mediatype.v1: scorecard+v1
+  operators.operatorframework.io.test.config.v1: tests/scorecard/
```
Comment (mvalarh) [inline operators/self-node-remediation/0.4.0/metadata/annotations.yaml:18]: Since there is no `tests/scorecard` direcory, please remove these lines

**Review by mvalarh — CHANGES_REQUESTED** [review]

File: operators/self-node-remediation/0.4.0/manifests/self-node-remediation.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+    node returning to a healthy state, the operator does allow surviving parts of
+    the cluster to assume the node has reached a safe state so that it’s workloads
+    can be automatically recovered. A prerequesuite for the self node remediation
+    operator is a functioning health detection system that implements the external
+    remediation API, such as Node Healthcheck Operator, Machine Healthcehck Controller
+    (in OCP or Cluster-API based clusters). Once a node/machine is unhealthy, the
+    detection system will create the SelfNodeRemediation CR, which triggers the self
+    node remediation operator. To get started, a user would need to create a SelfNodeRemediationTemplate,
+    and reference that CR with an NHC/MHC CR.
+  displayName: Self Node Remediation Operator
```
Comment (razo7) [inline operators/self-node-remediation/0.4.0/manifests/self-node-remediation.clusterserviceversion.yaml:116]: Change it to `Self Node Remediation Operator - Community Edition`

File: operators/self-node-remediation/0.4.0/metadata/annotations.yaml (hunk 1 above)
Comment (mshitrit) [reply operators/self-node-remediation/0.4.0/metadata/annotations.yaml:18]: Thanks, I've instead added tests/scorecard

**Pull request merged**